src/compile.rs

```diff
@@ -312,6 +326,81 @@ pub fn build_func_map<'a>(
     Ok(decl_map)
 }
 
+#[allow(clippy::collapsible_if)]
+pub fn validate_rules(statements: &BTreeSet<ValidatedStatement>) -> Result<(), CascadeErrors> {
+    let mut errors = CascadeErrors::new();
+    let mut fsc_rules: BTreeMap<&String, BTreeSet<&FileSystemContextRule>> = BTreeMap::new();
+
+    for statement in statements {
+        // Add all file system context rules to a new map to check for semi duplicates later
+        if let ValidatedStatement::FscRule(fs) = statement {
+            fsc_rules.entry(&fs.fs_name).or_default().insert(fs);
+        }
+    }
+
+    'key_loop: for (_, v) in fsc_rules {
+        // We only have 1 or 0 elements, thus we cannot have a semi duplicate
+        if v.len() <= 1 {
+            continue;
+        }
+        for rule in &v {
+            match rule.fscontext_type {
+                FSContextType::XAttr | FSContextType::Task | FSContextType::Trans => {
+                    // If we ever see a duplicate and one of them is xattr, task, or trans something is wrong
+                    // return an error and look for more.
+                    errors.append(CascadeErrors::from(InvalidFileSystemError::new(&format!(
+                        "Duplicate filesystem context.\n Found two different filesystem type declarations for filesystem: {}",
```

Comment:
I think we'll need to save the file with the FileSystemContextRule in call_to_fsc_rules() so that we can use it here.  We have the file at that point, so it should just be a matter of storing it in the struct.